File: DAL/Models/Order.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace DAL.Models
{
    public class Order : Entity
    {
        public String UserId { get; set; } = null!;
        public Int64 OrderNumber { get; set; }
        public Int64 TotalAmount { get; set; }
        public DateTime OrderDate { get; set; } = DateTime.Now;
        public OrderStatus Status { get; set; } = OrderStatus.Pending;
        public User User { get; set; } = null!;
        public IEnumerable<Product> Products { get; set; } = new List<Product>();

    }

    public enum OrderStatus
    {
        Pending,
        Processing,
        Shipped,
        Delivered,
        Cancelled
    }
}

File: webstoreprototype.client/src/app/interfaces/order.ts
import { Entity } from "./entity";
import { Product } from "./product";
import { User } from "./user";

export interface Order extends Entity {

    userId: string;
    orderNumber: number;
    productsCount: number;
    totalAmount: number;
    orderDate: Date;
    status: OrderStatus;
    user: User;
    products: Array<Product>;
}

enum OrderStatus{
    Pending,
    Processing,
    Shipped,
    Delivered,
    Cancelled
}
